Importers/Rhino/Speckle.Importers.JobProcessor/Blobs/ImportJobFile.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.Extensions.Logging;
+
+namespace Speckle.Importers.JobProcessor.Blobs;
+
+/// <summary>
+/// <see cref="IDisposable"/> wrapper around the downloaded file
+/// </summary>
+/// <remarks>
+/// This is an attempt to use a disposal pattern to ensure that downloaded file is deleted after its use.
+/// </remarks>
+/// <param name="logger"></param>
+/// <param name="file"></param>
+internal sealed class ImportJobFile(ILogger<ImportJobFile> logger, FileInfo file) : IDisposable
+{
+  public FileInfo FileInfo => file;
+
+  private void DeleteFile()
+  {
+    var dir = file.Directory;
+    try
+    {
+      file.Delete();
+    }
+    finally
+    {
+      dir?.Delete(true);
+    }
+  }
+
+  public void Dispose()
+  {
+    try
+    {
+      DeleteFile();
+      GC.SuppressFinalize(this);
+    }
+    catch (Exception ex)
+    {
+      logger.LogError(ex, "Failed to cleanup file");
+      throw;
+    }
+  }
+
+  ~ImportJobFile()
+  {
+    // Using the full unmanaged style disposal pattern means that the file will be deleted on finalise if dispose wasn't called already.
+    DeleteFile();
```

Comment:
Probably over kill, and not needed so much now Rhino is a seperate process... but, to provide some context, this is a standard C# pattern around "unmanaged" (i.e. c/c++) style disposal.
Ensuring that, even if `.Dispose` is not called (like in the case of a low level app crash), then the Finalizer will clean up any low level unmanaged resources.

Here, I'm not actually cleaning up any low level unmanaged resources, I'm deleting a file. But I figure that deleting this file is very important, so my hope is that this will be a belts + braces way to ensure it is.